### DISCUSSION TITLE
THE HYPIXEL WEBSITE WILL BE RIGHT BACK!

### DISCUSSION THREAD

**Discussion by justinrest** [HypixelDev/PublicAPI#526] [General]
![image](https://user-images.githubusercontent.com/92120745/160293577-fd4b463c-97e3-46f2-bf39-f24f95029dc5.png)

Randomly, the Hypixel API, yes, API Endpoints, will decide to have a "a gateway time-out error.", as an HTML site, not JSON.
This has been happening more and more frequently over the past few days.
It's like a 25% chance of it going to this page. 
It's not only me either, i've asked a few developers and they've said the same thing.

As far as i've read the Hypixel API only returns JSON.
![image](https://user-images.githubusercontent.com/92120745/160293536-7c500045-43b1-47eb-8f4c-37b78ea5802d.png)

Why is this happening? When is this going to be fixed?

**Comment by TheNullicorn** [comment]
It's been an issue for a while. For now, I believe the easiest way to get around this is checking the `content-type` header of the response. For actual API responses, it should contain `application/json`. For cloudflare error pages, it will contain `text/html`.

**Reply by justinrest** [reply]
I've already fixed it. This is mostly for the developers to fix it. Shouldn't even be a problem to start with.

**Comment by Picsou993** [comment]
You can also check the status code. You get a HTTP Status Code 200 when `result` is true and you can find them all in the doc at https://api.hypixel.net
